backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java

```diff
@@ -0,0 +1,134 @@
+
+    // Note: This method uses a stateless session
+    public Uni<Event> create(Event event) {
+        event.prePersist(); // This method must be called manually while using a StatelessSession.
+        return statelessSession.insert(event)
+                .replaceWith(event);
+    }
+
+    public Uni<List<Event>> get(String accountId, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeName,
+                                LocalDate startDate, LocalDate endDate, Integer limit, Integer offset, String sortBy) {
+        String hql = "SELECT DISTINCT e FROM Event e " +
+                "JOIN FETCH e.eventType et JOIN FETCH et.application a JOIN FETCH a.bundle b " +
+                "LEFT JOIN FETCH e.historyEntries he LEFT JOIN FETCH he.endpoint " +
+                "WHERE e.accountId = :accountId";
+
+        hql = addHqlConditions(hql, bundleIds, appIds, eventTypeName, startDate, endDate);
+
+        if (sortBy == null) {
+            hql += " ORDER BY e.created DESC";
+        } else {
+            Matcher sortByMatcher = SORT_BY_PATTERN.matcher(sortBy);
+            if (sortByMatcher.matches()) {
+                String sortField = getSortField(sortByMatcher.group(1));
+                String sortDirection = sortByMatcher.group(2);
+                hql += " ORDER BY " + sortField + " " + sortDirection + ", e.created DESC";
+            }
+        }
+
+        Mutiny.Query<Event> query = session.createQuery(hql, Event.class);
+        setQueryParams(query, accountId, bundleIds, appIds, eventTypeName, startDate, endDate);
+
+        if (limit != null) {
+            query.setMaxResults(limit);
+        }
+        if (offset != null) {
+            query.setFirstResult(offset);
+        }
+
+        return query.getResultList();
+    }
+
+    public Uni<Long> count(String accountId, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeName,
+                      LocalDate startDate, LocalDate endDate) {
+        String hql = "SELECT COUNT(*) FROM Event e JOIN e.eventType et JOIN et.application a JOIN a.bundle b " +
+                "WHERE e.accountId = :accountId";
+
+        hql = addHqlConditions(hql, bundleIds, appIds, eventTypeName, startDate, endDate);
+
+        Mutiny.Query<Long> query = session.createQuery(hql, Long.class);
+        setQueryParams(query, accountId, bundleIds, appIds, eventTypeName, startDate, endDate);
+
+        return query.getSingleResult();
+    }
+
+    private static String addHqlConditions(String hql, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeName,
+                                           LocalDate startDate, LocalDate endDate) {
+        if (bundleIds != null && !bundleIds.isEmpty()) {
+            hql += " AND b.id IN (:bundleIds)";
+        }
+        if (appIds != null && !appIds.isEmpty()) {
+            hql += " AND a.id IN (:appIds)";
+        }
+        if (eventTypeName != null) {
+            hql += " AND et.name LIKE :eventTypeName";
```

Comment:
I'm wondering if we should do the `LIKE` against `LOWER(et.displayName)` or both (in lower case) i.e. `" AND (LOWER(et.name) LIKE :eventTypeName OR LOWER(et.displayName) LIKE :eventTypeName)"`